feat(blog): May newsletter. (#6463)
jenkins-infra/jenkins.io: content/blog/2023/06/16/2023-06-16-jenkins-may-newsletter.adoc

Contributed by: author:markewaite[Mark Waite]

Several significant initiatives are already in progress within the Jenkins project.

Thanks to those who are leading the initiatives and thanks to those who are assisting with initiatives like:

* Prototype.js removal from Jenkins core and Jenkins plugins.
  Contributing guidelines are in the link:/blog/2023/05/12/removing-prototype-from-jenkins/[blog post].
  Detailed issue reports are available in the jira:JENKINS-70906[Jira epic].
  Progress reports for affected plugins are available in the link:https://docs.google.com/spreadsheets/d/1dpaKALZaK0_HIGy6ony3wnegr1frTg3u1lngG4KdoC8/edit?usp=sharing[tracking sheet].
* HTMLUnit 3 upgrade in the Jenkins test harness, Jenkins core, and in many Jenkins plugins
* Guava 32 upgrade

We're also excited to have additional efforts:

* Reduce core pull request evaluation time (and cost) with Launchable
* Use GitHub autolink for easier references to Jenkins Jira tickets

[[infrastructure]]
image:/images/post-images/2023/01/12/jenkins-newsletter/infrastructure.png[Infrastructure Update]
Contributed by: author:dduportal[Damien Duportal]

> * Cloud Cost Controls effort continued by optimizing resource usage:
> ** Decreased the AWS bill of $ 3,000 (14,000 -> 11,000).
> ** Decreased the Azure bill of $ 2,000 to (11,000 -> 9,000) despite adding resources.
> * Launchable is generally available for community developers on ci.jenkins.io.
> * Jenkins LTS `2.387.3` was deployed everywhere less than 24 hours after it was released.
> * Ubuntu `22.04` upgrade campaign (`18.04` is end of life) continued (6 more VMs, 5 VMs left).
> * Build workload migration to `ARM64`: internal tools.

Reviewer: ```suggestion
* The Cloud Cost Controls effort has continued by optimizing resource usage, resulting in:
** Decreased the AWS bill of $ 3,000 (14,000 -> 11,000).
** Decreased the Azure bill of $ 2,000 to (11,000 -> 9,000) despite adding resources.
* Launchable is now generally available for community developers on ci.jenkins.io.
* Jenkins LTS `2.387.3` was deployed everywhere less than 24 hours after it was released.
* Ubuntu `22.04` upgrade campaign (`18.04` is end of life) continued (6 more VMs, 5 VMs left).
* Build workload migration to `ARM64`: internal tools.
```
small syntax suggestions

[[platform]]
image:/images/post-images/2023/01/12/jenkins-newsletter/platform-modernization.png[Platform Modernization Update]

Contributed by: author:gounthar[Bruno Verachten]

* Deprecation
** Red Hat Enterprise Linux 7 (and derivatives) early end of life
*** To ensure a smooth transition, we are implementing several measures to inform users when an operating system is approaching its end of life.
These changes will be visible in upcoming releases and container images.
*** Key Dates:
**** May 28 - https://github.com/jenkinsci/jenkins/pull/7913[Pull request] merged
**** May 30 - link:/blog/2023/05/30/operating-system-end-of-life/[Blog post] and https://community.jenkins.io/t/end-of-life-operating-systems/7644[community topic]
**** May 30 - First weekly release containing the warning - `2.407`
**** Nov 16 - RHEL 7 support ends in Jenkins
* Ongoing work
** author:dduportal[Damien Duportal] is actively working on code factorization, specifically targeting a single repository for all agent images.
*** This will streamline maintenance tasks, such as fixing CVEs.
*** The first phase for JDK versions has already been completed, resulting in significantly reduced code size.

> *** Additionally, efforts are underway to merge two agents, aiming for synchronized release cycles. This change should not impact end users, except for the transition of the repository into an archive.

Reviewer: ```suggestion
*** Additionally, efforts are underway to merge two agents, aiming for synchronized release cycles.
This change should not impact end users, except for the transition of the repository into an archive.
```
small formatting suggestion

> ** We should provide more frequent updates on the development of Alpine images thanks to the use of `updatecli`.

Reviewer: ```suggestion
** We are able to provide more frequent updates on the development of Alpine images thanks to the use of `updatecli`.
```
small syntax suggestion

[[documentation]]
image:/images/post-images/2023/02/07/2023-02-07-jenkins-newsletter/documentation.png[Documentation Update]
Contributed by: author:markewaite[Mark Waite]

We're pleased to welcome several new documentation contributions, including our Google Summer of Code contributors:

* link:/blog/2023/05/26/ash-sxn-introduction-blog-post/[Ashutosh Saxena]
* link:/blog/2023/05/22/jagruti-introduction-blog-post/[Jagruti Tiwari]
* link:/blog/2023/05/18/harsh-ps-2003-introduction-blog-post/[Harsh Pratap Singh]
* link:/blog/2023/05/17/vandit1604-introduction-blog-post/[Vandit Singh]

We're very pleased that these new Jenkins contributors have seen the value of documentation and have submitted documentation improvements.

We also welcomed first time documentation contributors in May and are pleased to have their additions.
Thanks so much to our documentation contributors.

[[outreach]]
image:/images/post-images/2023/01/12/jenkins-newsletter/outreach-and-advocacy.png[Outreach and advocacy Update]

Contributed by: author:alyssat[Alyssa Tong]

image:/images/post-images/2023/06/16/2023-06-16-jenkins-may-newsletter/image2.png[image,width=839,Jenkins Contributor Awards 2023]

> Every year the community nominates, and votes for three outstanding difference makers in the Jenkins community: Most Valuable Advocate, Security MVP, and Most Valuable Contributor. Congratulations go to:

Reviewer: ```suggestion
Every year, the community nominates and votes for three outstanding difference makers in the Jenkins community: Most Valuable Advocate, Security MVP, and Most Valuable Contributor.
Congratulations go to:
```
formatting suggestion

* author:daniel-beck[Daniel Beck] - Security MVP.
* author:janfaracik[Jan Faracik] - Most Valuable Contributor.
* author:markewaite[Mark Waite] - Most Valuable Advocate.

Read what makes them link:/blog/2023/05/16/jenkins-2023-award-winners/[outstanding contributors].

Many THANKS and congratulations to all award nominees!

image:/images/post-images/2023/06/16/2023-06-16-jenkins-may-newsletter/image1.jpg[image,width=839,cdCon + GitOpsCon 2023]

> The link:https://cd.foundation/[Continuous Delivery Foundation (CDF)] hosted its fourth flagship event, cdCon, on May 8–9, 2023 in Vancouver, Canada as link:https://events.linuxfoundation.org/cdcon-gitopscon/[cdCon + GitOpsCon], co-organized with the link:https://cncf.io/[Cloud Native Computing Foundation (CNCF)].
> 
> Sessions from the most widely used CI/CD and GitOps technologies including the Jenkins community were there with project updates along with various talks from community members and users.
> 
> In case you missed it, below are the recorded Jenkins sessions at cdCon:
> 
> * link:https://youtu.be/9o5suZ0lRss[Fidelity’s Software Delivery Platform - Frictionless Approach to Achieve Autonomic DevOps & Enhanced Security/Compliance Practices] - Jamie Plower & Evan Elms, Fidelity Investments.
> * link:https://youtu.be/Yr3zNnbggfE[CI/CD for Data Building Dev/Test Data Environments with Open Source Stacks] - Vinodhini Duraisamy, Treeverse.
> * link:https://youtu.be/9QBORPTbkaY[Intentional and Unintentional Compromises in Test Automation] - Mark Waite, Jenkins.
> * link:https://youtu.be/vz7_rAQz_9c[The Graduated Panel] https://cdcongitopscon2023.sched.com/?iframe=yes&w=100%&sidebar=yes&bg=no#[- Dan Garfield, Codefresh; Priyanka Ravi, Weaveworks; Mark Waite, CloudBees; Andrea Frittoli, IBM & Moderated by Lori Lorusso, JFrog].

Reviewer: ```suggestion
The link:https://cd.foundation/[Continuous Delivery Foundation (CDF)] hosted its fourth flagship event, cdCon, on May 8 – 9, 2023 in Vancouver, Canada as link:https://events.linuxfoundation.org/cdcon-gitopscon/[cdCon + GitOpsCon], co-organized with the link:https://cncf.io/[Cloud Native Computing Foundation (CNCF)].

Sessions from the most widely used CI/CD and GitOps technologies, including the Jenkins community, were there with project updates along with various talks from community members and users.

In case you missed it, below are the recorded Jenkins sessions at cdCon:

* link:https://youtu.be/9o5suZ0lRss[Fidelity’s Software Delivery Platform - Frictionless Approach to Achieve Autonomic DevOps & Enhanced Security/Compliance Practices] - Jamie Plower & Evan Elms, Fidelity Investments.
* link:https://youtu.be/Yr3zNnbggfE[CI/CD for Data Building Dev/Test Data Environments with Open Source Stacks] - Vinodhini Duraisamy, Treeverse.
* link:https://youtu.be/9QBORPTbkaY[Intentional and Unintentional Compromises in Test Automation] - Mark Waite, Jenkins.
* link:https://youtu.be/vz7_rAQz_9c[The Graduated Panel] https://cdcongitopscon2023.sched.com/?iframe=yes&w=100%&sidebar=yes&bg=no#[- Dan Garfield, Codefresh; Priyanka Ravi, Weaveworks; Mark Waite, CloudBees; Andrea Frittoli, IBM & Moderated by Lori Lorusso, JFrog].
```
small formatting and punctuation suggestion
